Restore `Commit` to the ABCI++ spec, and other late modifications (#8796)
tendermint/tendermint: spec/abci++/abci++_app_requirements_002_draft.md

Full execution of blocks at `PrepareProposal` time stands on Tendermint's critical path. Thus,
Requirement 3 ensures the Application will set a value for `TimeoutPropose` such that the time it takes
to fully execute blocks in `PrepareProposal` does not interfere with Tendermint's propose timer.
Note that violation of Requirement 3 may just lead to further rounds, but not to Tendermint

> liveness issues.

Author: Thanks, I like it

Setting this parameter to `false` (the default) causes Tendermint to wait
for the full commit timeout configured in `TimeoutParams.Commit`.

##### ABCIParams.VoteExtensionsEnableHeight

This configures the height at which vote extensions become mandatory.
The default value is 0, and means that so far the Application has not
set the height yet.
When the configured height `H` is reached, `PrepareProposal` will not
include vote extensions yet, but `ExtendVote` and `VerifyVoteExtension` will
be called. Then, when reaching height `H+1`, `PrepareProposal` will
include the vote extensions from height `H`. From this point on, vote extensions
are mandatory although the application can provide 0-length extensions.

Must always be set to a future height. Once set to a value different from

> 0, its value must not be changed.

Reviewer: Maybe just to hammer the point in say that once VoteExtensions are enabled they can't be disabled

Author: Added it in the previous paragraph

Setting this parameter to `false` (the default) causes Tendermint to wait
for the full commit timeout configured in `TimeoutParams.Commit`.

##### ABCIParams.VoteExtensionsEnableHeight

This configures the height at which vote extensions become mandatory.
The default value is 0, and means that so far the Application has not
set the height yet.
When the configured height `H` is reached, `PrepareProposal` will not
include vote extensions yet, but `ExtendVote` and `VerifyVoteExtension` will
be called. Then, when reaching height `H+1`, `PrepareProposal` will

> include the vote extensions from height `H`. From this point on, vote extensions

Reviewer 2: ```suggestion
include the vote extensions from height `H`. For all heights after `H`, vote extensions
```
I think 'this point on' is a bit more vague than we should be in this document.

> are mandatory and cannot be disabled, although the application can provide 0-length

Reviewer 2: Should we define what mandatory means? i.e. that chains will not make progress if votes don't include extensions.
